query/reverse.js

```diff
@@ -65,13 +65,24 @@ function generateQuery( clean ){
   // where point.lan/point.lon are provided in the
   // absense of boundary.circle.lat/boundary.circle.lon
   if( check.number(clean['boundary.circle.lat']) &&
-      check.number(clean['boundary.circle.lon']) &&
-      check.number(clean['boundary.circle.radius']) ){
-    vs.set({
-      'boundary:circle:lat': clean['boundary.circle.lat'],
-      'boundary:circle:lon': clean['boundary.circle.lon'],
-      'boundary:circle:radius': clean['boundary.circle.radius'] + 'km'
-    });
+      check.number(clean['boundary.circle.lon']) ){
+
+        vs.set({
+          'boundary:circle:lat': clean['boundary.circle.lat'],
+          'boundary:circle:lon': clean['boundary.circle.lon']
+        });
+
+        if (check.undefined(clean['boundary.circle.radius'])){
+          // for coarse reverse when boundary circle radius is undefined
```

Comment:
this could also be the case if it's a non-coarse reverse query but the query didn't specify a boundary.circle.radius, right?